src/docfx/Errors.cs

```diff
@@ -4,174 +4,390 @@
+        /// </summary>
         public static Error FileNotFound(string relativeTo, string path)
             => new Error(ErrorLevel.Warning, "file-not-found", $"Cannot find file '{path}' relative to '{relativeTo}'", relativeTo);
 
+        /// <summary>
+        /// Failed to resolve uid defined by [link](xref:uid) syntax.
+        /// </summary>
         public static Error UidNotFound(Document file, string uid, string rawXref)
             => new Error(ErrorLevel.Warning, "uid-not-found", $"Cannot find uid '{uid}' using xref '{rawXref}'", file.ToString());
 
+        /// <summary>
+        /// File contains git merge conflict.
+        /// </summary>
         public static Error MergeConflict(string file)
             => new Error(ErrorLevel.Error, "merge-conflict", "File contains merge conflict", file);
 
+        /// <summary>
+        /// Failed to resolve uid defined by @ syntax.
+        /// </summary>
         public static Error AtUidNotFound(Document file, string uid, string rawXref)
             => new Error(ErrorLevel.Info, "at-uid-not-found", $"Cannot find uid '{uid}' using xref '{rawXref}'", file.ToString());
 
+        /// <summary>
+        /// Files published to the same url have no monikers or share common monikers.
+        /// </summary>
         public static Error PublishUrlConflict(string url, IEnumerable<Document> files, IEnumerable<string> conflictMonikers)
         {
             var message = !conflictMonikers.Contains("NONE_VERSION") ? $" of the same version({Join(conflictMonikers)})" : null;
             return new Error(ErrorLevel.Error, "publish-url-conflict", $"Two or more files{message} publish to the same url '{url}': {Join(files, file => file.ContentType == ContentType.Redirection ? $"{file} <redirection>" : file.ToString())}");
         }
 
+        /// <summary>
+        /// Inclusion is a redirections entry like [!INCLUDE[](redirect.md)]
+        /// </summary>
         public static Error IncludeRedirection(Document relativeTo, string path)
             => new Error(ErrorLevel.Warning, "include-is-redirection", $"Referenced inclusion {path} relative to '{relativeTo}' shouldn't belong to redirections", relativeTo.ToString());
 
+        /// <summary>
+        /// More than one files are resolved to the same output path.
+        /// Examples:
+        ///   - in config's redirections section, defined an entry key that's also a file in build scope
+        ///   - different file extension with same filename, like `Toc.yml and `Toc.md`
+        /// </summary>
         public static Error OutputPathConflict(string path, IEnumerable<Document> files)
             => new Error(ErrorLevel.Error, "output-path-conflict", $"Two or more files output to the same path '{path}': {Join(files, file => file.ContentType == ContentType.Redirection ? $"{file} <redirection>" : file.ToString())}");
 
+        /// <summary>
+        /// Multiple files defined in config's redirections are redirected to the same url,
```

Comment:
Does the condition need to **AND** "with `redirect_document_id` set to `true`" (I am not sure of the config name...)